controllers/keyword.go

```diff
@@ -0,0 +1,54 @@
+package controllers
+
+import (
+	"encoding/csv"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/gin-gonic/gin"
+	html "github.com/gutakk/go-google-scraper/helpers/html"
+	"github.com/gutakk/go-google-scraper/models"
+)
+
+const (
+	keywordTitle = "Keyword"
+	keywordView  = "keyword"
+)
+
+type KeywordController struct{}
+
+func (k *KeywordController) applyRoutes(engine *gin.Engine) {
+	engine.GET("/keyword", k.displayKeyword)
+	engine.POST("/keyword", k.uploadKeyword)
+}
+
+func (k *KeywordController) displayKeyword(c *gin.Context) {
+	html.RenderWithFlash(c, http.StatusOK, keywordView, keywordTitle, nil)
+}
+
+func (k *KeywordController) uploadKeyword(c *gin.Context) {
+	file, _ := c.FormFile("file")
+
+	// Validate if file is CSV type
+	if err := models.ValidateFileType(file.Header["Content-Type"][0]); err != nil {
+		html.RenderWithError(c, http.StatusBadRequest, keywordView, keywordTitle, err, nil)
+	}
+
+	filename := "dist/" + filepath.Base(file.Filename)
+	_ = c.SaveUploadedFile(file, filename)
+	csvfile, _ := os.Open(filename)
+	r := csv.NewReader(csvfile)
+	record, _ := r.ReadAll()
```

Comment:
Line 38 -> 42 should also be moved outside of the controller for the same reasons I gave you before: it's the business domain 💍 There needs to be an API to perform CSV upload so the same logic would be needed. Encapsulating it into a new object would make it reusable e.g. `keywordFile` and a method `keywordFile.upload()`.

Great to move `ValidateCSVLength` and `SaveKeywords` to the keyword model 💪

Reply:
Fixed in 45e248a014c3d66aa52efa2092fa3bc3f97eb9c4

Reply:
There could also be another model for KeyworsdFile specifically for the file validation 💡